src/hours/hours-grid/HoursGrid.component.tsx

```diff
@@ -98,31 +124,57 @@ const HoursGrid = ({
                                     );
                                 })}
                                 <TableCell>
-                                    {Boolean(readOnly) ? (
-                                        row.explanation
-                                    ) : (
-                                        <TextField
-                                            id="explanation"
-                                            inputProps={{
-                                                day: row.day,
-                                            }}
-                                            value={row.explanation}
-                                            onBlur={(event) =>
-                                                handleChange(
-                                                    event.target.value,
-                                                    "explanation",
-                                                    row.day,
-                                                )
+                                    <Button
+                                        onClick={(event) => {
+                                            openComment(row.explanation, index);
+                                        }}
+                                        id={"commentButton-" + index}
+                                    >
+                                        <ChatBubbleOutlineIcon
+                                            color={
+                                                row.explanation === ""
+                                                    ? "disabled"
+                                                    : "primary"
                                             }
                                         />
-                                    )}
+                                    </Button>
                                 </TableCell>
                             </TableRow>
                         );
                     })}
                 </TableBody>
                 <HoursFooter expandColumns={expandColumns} days={days} />
             </Table>
+
+            <Dialog
```

Comment:
<img width="1552" alt="Screenshot 2020-05-12 at 09 50 38" src="https://user-images.githubusercontent.com/39329053/81654973-35cd1900-9436-11ea-89c4-95f07c3b3323.png">

Wat ik merk is dat wanneer je typt in de comment box het lang duurt voordat je je tekst ziet (meerdere secondes). Als je naar de performance kijkt, zie je dat het hele HoursGrid opnieuw rendert als je een verandering maakt. Ik vermoed dat dat komt omdat het Dialog component in het HoursGrid.tsx zit. Ik hoop dat als je hier een eigen component van maakt de update tijd versnelt en alleen dit component opnieuw renderd bij een verandering. Volgens mij kan dat goed en is readonly dan de enige prop die doorgegeven hoeft te worden.

Reply:
Ik heb hem eerst in een losse component gehad, maar ik kreeg juist moeite met het updaten van het veld bij het meerdere keren openen van het dialog. Maar ik ga er nog eens naar kijken.